test/ts/unit/simple_interest_terms_contract.ts

```diff
@@ -0,0 +1,237 @@
+
+                before(async () => {
+                    await router.repay.sendTransactionAsync(
+                        ARBITRARY_AGREEMENT_ID,
+                        Units.ether(1),
+                        NULL_ADDRESS // this is not the address it's expecting.
+                    );
+                });
+
+                it("should not record the repayment in any capacity", async () => {
+                  await expect(termsContract.getValueRepaidToDate.callAsync(
+                    ARBITRARY_AGREEMENT_ID)).to.eventually.equal(Units.ether(0));
+                });
+            });
+
+            describe("...with terms contract's expected `tokenAddress`", () => {
+
+                before(async () => {
+                    await router.repay.sendTransactionAsync(
+                        ARBITRARY_AGREEMENT_ID,
+                        Units.ether(1),
+                        mockToken.address // this is the address it's expecting.
+                    );
+                });
+
+                it("should record the repayment", async () => {
+                  await expect(termsContract.getValueRepaidToDate.callAsync(
+                    ARBITRARY_AGREEMENT_ID)).to.eventually.equal(Units.ether(1));
+                });
+
+            });
+        });
+    });
+
+    describe("#getValueRepaidToDate", () => {
+
+        describe("non-existent debt agreement", () => {
+            /*
+            Question(kayvon): as I understand it, the contract wouldn't throw,
+            but just return 0. Do we want to change this implementation?
+            */
+            it("should throw");
+        });
+
+        describe("extant debt agreement", () => {
+
+          describe("at time N, debtor has made an initial payment of 1 ether", () => {
+
+              before(async () => {
+                  await router.repay.sendTransactionAsync(
+                      ARBITRARY_AGREEMENT_ID,
+                      Units.ether(1),
+                      mockToken.address
+                  );
+              });
+
+
+              it("returns 1 ether", async () => {
+                await expect(termsContract.getValueRepaidToDate.callAsync(
+                  ARBITRARY_AGREEMENT_ID)).to.eventually.equal(Units.ether(1));
+              });
+
+          });
+
+          describe("at time M, debtor has made an additional payment of 1 ether", () => {
+
+              before(async () => {
+                  // initial payment 1 ether.
+                  await router.repay.sendTransactionAsync(
+                      ARBITRARY_AGREEMENT_ID,
+                      Units.ether(1),
+                      mockToken.address
+                  );
+
+                  // second payment of 1 ether.
+                  await router.repay.sendTransactionAsync(
+                      ARBITRARY_AGREEMENT_ID,
+                      Units.ether(1),
+                      mockToken.address // this is not the address it's expecting.
```

Comment:
Au contraire -- this _is_ the address it's expecting

Reply:
yeah, copy and paste served me poorly here.